### PR TITLE
SF3 compatibility

### PR DESCRIPTION
**Pull request by nicodmf**
All is in the title, this is not the first PR, but this is an important commit and as SF3 has more than one year, this seems to me than this should integrated shortly.

### CODE REVIEW

**Comment by greg0ire** [comment]
This is just like #1115, with the same problems, unresolved.

File: Admin/BaseMediaAdmin.php [hunk 1]
```diff
@@ -106,7 +106,7 @@ public function getNewInstance()
 
         if ($this->hasRequest()) {
             if ($this->getRequest()->isMethod('POST')) {
-                $media->setProviderName($this->getRequest()->get(sprintf('%s[providerName]', $this->getUniqid()), null, true));
+                $media->setProviderName($this->getRequest()->get(sprintf('%s', $this->getUniqid()), null, true)['providerName']);
```
Comment (greg0ire) [inline Admin/BaseMediaAdmin.php]: The final code should look like this : 

```
if (method_exists('Symfony\Component\HttpFoundation\JsonResponse', 'transformJsonError')) {
    // NEXT_MAJOR remove this block when dropping sf < 2.8 compatibility
    $media->setProviderName($this->getRequest()->get(sprintf('%s[providerName]', $this->getUniqid()), null, true));
} else {
    $media->setProviderName(
        $this->getRequest()->get($this->getUniqid()['providerName'])
    );
}
```

File: Resources/config/validation.xml [hunk 2]
```diff
@@ -1,11 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping         http://symfony.com/schema/dic/services/constraint-mapping-1.0.xsd">
     <class name="Sonata\MediaBundle\Model\Media">
-        <constraint name="Callback">
-            <option name="methods">
-                <value>isStatusErroneous</value>
-            </option>
-        </constraint>
+        <constraint name="Callback">isStatusErroneous</constraint>
```
Comment (greg0ire) [inline Resources/config/validation.xml]: Does this work with symfony 2.3 ?

File: Admin/GalleryAdmin.php [hunk 3]
```diff
@@ -128,7 +128,6 @@ protected function configureFormFields(FormMapper $formMapper)
             ->end()
             ->with('Gallery')
                 ->add('galleryHasMedias', 'sonata_type_collection', array(
-                        'cascade_validation' => true,
```
Comment (greg0ire) [inline Admin/GalleryAdmin.php]: Why remove this?

Comment (greg0ire) [inline Admin/GalleryAdmin.php]: cascade_validation was deprecated in 2.8.0 for anyone who is wondering.

Comment (greg0ire) [inline Admin/GalleryAdmin.php]: It has to be replaced with the `Valid` constraint

**Comment by greg0ire** [comment]
This has been properly fixed in #1191 if you want to have a look.

**Comment by nicodmf** [comment]
Thanks :)

**Pull request closed without merging**